lib/Parse/ParseExpr.cpp

```diff
@@ -1386,6 +1386,55 @@ ExprResult Parser::ParseCastExpression(bool isUnaryExpression,
   // are compiling for OpenCL, we need to return an error as this implies
   // that the address of the function is being taken, which is illegal in CL.
 
+  // With primary-expression, before parsing postfix-expression, check to make
+  // sure that Expr is declRefExpr of generic function with _For_any specifier
+  Expr* resExpr = Res.get();
+  if (resExpr != NULL && isa<DeclRefExpr>(resExpr)) {
+    DeclRefExpr* declRef = dyn_cast<DeclRefExpr>(resExpr);
+    ValueDecl* resDecl = declRef->getDecl();
+    if (resDecl != NULL && isa<FunctionDecl>(resDecl)) {
```

Comment:
Please use nullptr, not NULL.   You can also just use `if (resDecl && isa<FunctionDecl>...`